print-apps/oereb/topicotherlegend.jrxml

```diff
@@ -7,29 +7,51 @@
 	<detail>
 		<band height="17" splitType="Stretch">
 			<property name="com.jaspersoft.studio.unit.height" value="pixel"/>
+			<printWhenExpression><![CDATA[!($F{LegendText}.equals( "" ) || $F{LegendText} == null)]]></printWhenExpression>
 			<image>
 				<reportElement x="0" y="2" width="18" height="9" isRemoveLineWhenBlank="true" uuid="b9ef005c-0354-4c25-92dc-0121e89d814c">
 					<property name="com.jaspersoft.studio.unit.width" value="mm"/>
 					<property name="com.jaspersoft.studio.unit.height" value="pixel"/>
 					<property name="com.jaspersoft.studio.unit.x" value="pixel"/>
 					<property name="com.jaspersoft.studio.unit.y" value="pixel"/>
+					<printWhenExpression><![CDATA[!($F{SymbolRef}.equals( "" ) || $F{SymbolRef} == null)]]></printWhenExpression>
 				</reportElement>
 				<box>
 					<pen lineWidth="0.2"/>
 				</box>
-				<imageExpression><![CDATA[($F{SymbolRef}.equals("") || $F{SymbolRef} == null) ? null : $F{SymbolRef}]]></imageExpression>
+				<imageExpression><![CDATA[$F{SymbolRef}]]></imageExpression>
 			</image>
 			<textField isStretchWithOverflow="true" isBlankWhenNull="true">
-				<reportElement x="27" y="0" width="272" height="17" uuid="5f9a91b3-6ee9-4982-a8db-35245b462c2a">
+				<reportElement x="27" y="0" width="272" height="17" isRemoveLineWhenBlank="true" uuid="5f9a91b3-6ee9-4982-a8db-35245b462c2a">
 					<property name="com.jaspersoft.studio.unit.x" value="pixel"/>
 					<property name="com.jaspersoft.studio.unit.y" value="pixel"/>
 					<property name="com.jaspersoft.studio.unit.height" value="pixel"/>
+					<printWhenExpression><![CDATA[!($F{LegendText}.equals( "" ) || $F{LegendText} == null)]]></printWhenExpression>
```

Comment:
This actually is now redundant when you add this same in [line 10](https://github.com/openoereb/pyramid_oereb_mfp/pull/33/files#diff-b05657e039c84cd17ed93d53ad947965R10) the whole band will not be printed if the condition is fulfilled. So you can remove this line.

Reply:
well spotted, thanks